be/src/exprs/map_apply_expr.cpp

```diff
@@ -0,0 +1,141 @@
+            auto nullable = std::dynamic_pointer_cast<NullableColumn>(child_col);
+            DCHECK(nullable != nullptr);
+            data_column = nullable->data_column();
+            // empty null map with non-empty elements
+            data_column->empty_null_in_complex_column(
+                    nullable->null_column()->get_data(),
+                    std::dynamic_pointer_cast<MapColumn>(data_column)->offsets_column()->get_data());
+            if (input_null_map) {
+                input_null_map =
+                        FunctionHelper::union_null_column(nullable->null_column(), input_null_map); // merge null
+            } else {
+                input_null_map = nullable->null_column();
+            }
+        }
+        DCHECK(data_column->is_map());
+        auto cur_map = std::dynamic_pointer_cast<MapColumn>(data_column);
+
+        if (input_map == nullptr) {
+            input_map = cur_map;
+        } else {
+            if (!ColumnHelper::offsets_equal(cur_map->offsets_column(), input_map->offsets_column())) {
+                throw std::runtime_error("Input map element's size are not equal in map_apply().");
+            }
+        }
+        input_columns.push_back(cur_map->keys_column());
+        input_columns.push_back(cur_map->values_column());
+    }
+    // step 2: construct a new chunk to evaluate the lambda expression, output a map column without warping null info.
+    ColumnPtr column = nullptr;
+    if (input_map->keys_column()->size() == 0) { // map is empty
+        column = ColumnHelper::create_column(type(), false);
+    } else {
+        auto cur_chunk = std::make_shared<Chunk>();
+        // put all arguments into the new chunk
+        std::vector<SlotId> arguments_ids;
+        auto lambda_func = dynamic_cast<LambdaFunction*>(_children[0]);
+        int argument_num = lambda_func->get_lambda_arguments_ids(&arguments_ids);
+        DCHECK(argument_num == input_columns.size());
+        for (int i = 0; i < argument_num; ++i) {
+            cur_chunk->append_column(input_columns[i], arguments_ids[i]); // column ref
+        }
+        // put captured columns into the new chunk aligning with the first map's offsets
+        std::vector<SlotId> slot_ids;
+        _children[0]->get_slot_ids(&slot_ids);
+        for (auto id : slot_ids) {
+            DCHECK(id > 0);
+            auto captured = chunk->get_column_by_slot_id(id);
+            if (captured->size() < input_map->size()) {
+                throw std::runtime_error(fmt::format("The size of the captured column {} is less than map's size.",
+                                                     captured->get_name()));
+            }
+            cur_chunk->append_column(captured->replicate(input_map->offsets_column()->get_data()), id);
+        }
+        // evaluate the lambda expression
+        if (cur_chunk->num_rows() <= chunk->num_rows() * 8) {
+            column = EVALUATE_NULL_IF_ERROR(context, _children[0], cur_chunk.get());
```

Comment:
context->evaluate(child[0],chunk);

Reply:
why? `EVALUATE_NULL_IF_ERROR` is more commonly used.